Services/GEV/DecentralTrainings/classes/class.gevDecentralTrainingCreationRequest.php

```diff
@@ -87,89 +79,101 @@ public function __construct( gevDecentralTrainingCreationRequestDB $db
 		$this->finished_ts = $a_finished_ts;
 		$this->created_obj_id = $a_created_obj_id;
 	}
-	
-	public function requestId() {
+
+	public function requestId()
+	{
 		return $this->request_id;
 	}
-	
-	public function userId() {
+
+	public function userId()
+	{
 		return $this->user_id;
 	}
-	
-	public function sessionId() {
+
+	public function sessionId()
+	{
 		return $this->session_id;
 	}
-	
-	public function templateObjId() {
+
+	public function templateObjId()
+	{
 		return $this->template_obj_id;
 	}
-	
-	public function settings() {
+
+	public function settings()
+	{
 		return $this->settings;
 	}
 
-	public function setSettings($settings) {
+	public function setSettings($settings)
+	{
 		$this->settings = $settings;
 	}
-	
-	public function trainerIds() {
+
+	public function trainerIds()
+	{
 		return $this->trainer_ids;
 	}
-	
-	public function requestedTS() {
+
+	public function requestedTS()
+	{
 		return $this->requested_ts;
 	}
-	
-	public function finishedTS() {
+
+	public function finishedTS()
+	{
 		return $this->finished_ts;
 	}
-	
-	public function createdObjId() {
+
+	public function createdObjId()
+	{
 		return $this->created_obj_id;
 	}
-	
-	public function request() {
-		$this->requested_ts = new ilDateTime(time(),IL_CAL_UNIX);
+
+	public function request()
+	{
+		$this->requested_ts = new ilDateTime(time(), IL_CAL_UNIX);
 		$this->session_id = $this->getNewSessionId();
 		if ($this->request_id === null) {
 			$this->request_id = $this->db->createRequest($this);
-		}
-		else {
+		} else {
 			$this->db->updateRequest($this);
 		}
 	}
-	
-	public function save() {
+
+	public function save()
+	{
 		if ($this->request_id === null) {
 			$this->request_id = $this->db->createRequest($this);
-		}
-		else {
+		} else {
 			$this->db->updateRequest($this);
 		}
 	}
-	
-	public function delete() {
+
+	public function delete()
+	{
 		$this->db->deleteRequest($this);
 		$this->request_id = null;
 	}
-	
-	public function run() {
+
+	public function run()
+	{
 		if ($this->finished_ts !== null) {
 			$this->throwException("Request already finished.");
 		}
-		
+
 		if ($this->isSessionExpired()) {
 			$this->throwException("Session '".$this->session_id."' is expired.");
 		}
-		
+
 		$rbacsystem = $this->getRBACSystem();
-		
+
 		$this->checkPermissionToCreateTrainingForTrainers();
 
 		$src_utils = $this->getCourseUtils($this->template_obj_id);
-		
+
 		$trgt_ref_id = $this->cloneTemplate($src_utils->getCourse());
```

Comment:
Verstehe ich das Richtig, dass der SOAP clone Vorgang hier einsetzt? Wäre es nicht gescheiter direkt hiernach zu warten, um ein aktuelles objekt in trgt_utils und trgt_crs weiter unten zu erhalten?

Reply:
done